docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,206 @@
+
+In the following scenario, we assume the EdgeX services are running in Docker environments,
+and thus the examples are given in terms of Docker-compose ways.  It should not be much or bigger
+difference for `snap` running environment to apply the same steps or concepts if found to do so.
+
+If a user wants to configure and set up an add-on service, e.g. `device-camera`,
+he can do the steps of guidelines as follows to achieve that:
+
+## Make add-on services security-bootstrapper compatible
+
+To use the Docker entrypoint scripts of gating mechanism from `security-bootstrapper`,
+the Dockerfile of `device-camera` should inherit shell scripting capability like `alpine`-based
+as the base Docker image and should install `dumb-init`(see details in
+[Why you need an init system](https://github.com/Yelp/dumb-init#why-you-need-an-init-system))
+via `apk add --update` command.
+
+Dockerfile example using alpine-base image and add `dumb-init`:
+
+```dockerfile
+......
+FROM alpine:3.12
+
+# dumb-init needed for injected secure bootstrapping entrypoint script when run in secure mode.
+RUN apk add --update --no-cache dumb-init
+......
+
+```
+
+and then in the service itself should add `/edgex-init/ready_to_run_wait_install.sh` as the entrypoint script
+for the service in gating fashion and related Docker volumes from `edgex-init` and secretstore token which
+will be outlined in the next section.
+
+A good example of this will be like `app-serive-rules`:
+
+```yaml
+...
+  app-service-rules:
+    entrypoint: ["/edgex-init/ready_to_run_wait_install.sh"]
+    command: "/app-service-configurable ${DEFAULT_EDGEX_RUN_CMD_PARMS}"
+    volumes:
+      - edgex-init:/edgex-init:ro,z
+      - /tmp/edgex/secrets/app-rules-engine:/tmp/edgex/secrets/app-rules-engine:ro,z
+    depends_on:
+      - security-bootstrapper
+...
+
+```
+
+## Configure the secret store to use
+
+Make sure the TOML configuration file of add-on service like `device-camera` contains
+the proper `[SecretStore]` section.
+
+Example:
+
+```toml
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = 'device-camera/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/device-camera/secrets-token.json'
+  [SecretStore.Authentication]
+  AuthType = 'X-Vault-Token'
+```
+
+and then in the EdgeX service `secretstore-setup` environment section of `docker-compose` file to add
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
+...
+
+```
+
+With that, `secretstore-setup` then can generate secretstore token from `Vault` and store it in
+the `TokenFile` path specified in the TOML configuration file like the above example.
+
+## Configure the secure Redis message bus to use
```

Comment:
ok since you suggest changing the section title, now this makes sense to the section I outline below.  However, this creates some confusion of securing Redis message bus for user.  Why we want to refer all of that to this and call it securing Redis message bus. If I were the user, I got lost about the relationship to convey between the "securing redis message bus" and this whole page.

Thoughts?

Reply:
We are not `securing` the Redis MessageBus. 
We are explaining the steps an add-on service needs to do to use the already `secured` MessageBus. This is just one of those steps. 
If you want to add numbers to the steps, we could reference them in the Secure MessageBus document, i.e. Steps 1 & 2 from this document are required for the add-on service to connect to the secure MessageBus.

Reply:
ok, thanks for the explanation.  the number steps are not working well in this documentation, I 'll leave as it is in terms of sections.   
So in the "Secure Message Bus" documentation we refer to these documentation and all the sections are steps here to configure to use "secure message bus" so to speak. 
I see that in the other documentation but not here hence kind of getting lost originally.  
maybe adding some sentences at the very beginning (the introduction section of this doc) to mention that probably would make that connection better.

Reply:
This documentation is the big picture of adding an ad-on service. The steps needed for connecting to the secure MessageBus are just creating the secret store and adding the `redisdb` known secret (which is not needed for those not connecting to the bus or needing the DB),

Reply:
sounds good, we can add what you said in this section which is actually an important concept convey to user

Reply:
changed